src/google/protobuf/compiler/objectivec/objectivec_helpers.cc

```diff
@@ -1267,6 +1269,61 @@ bool ValidateObjCClassPrefix(
     return true;
   }
 
+  // When the prefix is non empty, check it against the expected entries.
+  if (!prefix.empty() && have_expected_prefix_file) {
+    // For a non empty prefix, look for any other package that uses the prefix.
+    std::string other_package_for_prefix;
+    for (std::map<std::string, std::string>::const_iterator i =
```

Comment:
std::string other_package_for_prefix;
auto iter = expected_package_prefixes.find(prefix);
if (iter != expected_package_prefixes.end()) {
  other_package_for_prefix = iter->second;
}

Reply:
Doesn't `find` look up by key?  The map is _package_ -> _prefix_, so I think the only option is what I'm already doing, iterate to find the _prefix_ and make note of the _package_ that uses it?

Reply:
ouch.. got you. Totally missed that. Thanks.